src/main/java/com/cinema/dao/impl/MovieDaoImpl.java

```diff
@@ -0,0 +1,41 @@
+package com.cinema.dao.impl;
+
+import com.cinema.dao.MovieDao;
+import com.cinema.lib.Dao;
+import com.cinema.model.Movie;
+import com.cinema.util.HibernateUtil;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+
+    @Override
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
```

Comment:
We should use the try-finally block in the non-read operations, the checklist says